sklearn/linear_model/_stochastic_gradient.py

```diff
@@ -55,6 +55,7 @@ def __init__(self, estimator, X_val, y_val, sample_weight_val,
                  classes=None):
         self.estimator = clone(estimator)
         self.estimator.t_ = 1  # to pass check_is_fitted
+        self.estimator.n_features_in_ = X_val.shape[1]
```

Comment:
That's one more reason not to validate against `n_features_in_` when it's missing.